core/utils_test.py

```diff
@@ -794,3 +795,144 @@ def test_url_open(self) -> None:
         self.assertEqual(response.getcode(), 200) # type: ignore[attr-defined]
         self.assertEqual(
             response.url, 'http://www.google.com') # type: ignore[attr-defined]
+
+    def test_get_random_int(self) -> None:
+        self.assertLess(utils.get_random_int(5), 5)
+        self.assertGreaterEqual(utils.get_random_int(5), 0)
+        self.assertLess(utils.get_random_int(True), 1)
+        self.assertGreaterEqual(utils.get_random_int(True), 0)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only positive integers allowed'):
+            utils.get_random_int(-1)
+
+    def test_get_random_choice(self) -> None:
+        list_instance = [1, 5, 9, 11, 15]
+        list_instance2: List[str] = []
+        self.assertIn(utils.get_random_choice(list_instance), list_instance)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only non-empty lists allowed'):
+            utils.get_random_choice(list_instance2)
+
+    def test_get_human_readable_time_string(self) -> None:
+        self.assertEqual(
+            'December 12 06:42:12',
+            utils.get_human_readable_time_string(944980932342.38)
+        )
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Time cannot be negative'):
+            utils.get_human_readable_time_string(-1.42)
+
+    def test_generate_new_session_id(self) -> None:
+        test_string = utils.generate_new_session_id()
+        self.assertEqual(24, len(test_string))
+        self.assertIsInstance(test_string, str)
+        test_string_utf8 = test_string.encode('utf-8')
+        test_string_utf16 = test_string.encode('utf-16')
+        self.assertEqual(
+            len(test_string), len(test_string_utf8))
+        self.assertNotEqual(
+            len(test_string), len(test_string_utf16))
+
+    def test_require_valid_name_with_incorrect_input(self) -> None:
+        """ADD TEST ESCAPE SEQUENCE CHARACTERS CHECK."""
```

Comment:
Now that you've added a test for this, we don't need this comment anymore, right?

Reply:
Done